pkg/node/discovery/utils.go

```diff
@@ -35,24 +36,30 @@ func (d *Discovery) verifyDriveMount(existingDrive *directcsi.DirectCSIDrive) er
 	driveMounter := &sys.DefaultDriveMounter{}
 	switch existingDrive.Status.DriveStatus {
 	case directcsi.DriveStatusInUse, directcsi.DriveStatusReady:
-		directCSIDevice := sys.GetDirectCSIPath(existingDrive.Status.FilesystemUUID)
+		mountSource := sys.GetDirectCSIPath(existingDrive.Status.FilesystemUUID)
 		// verify and fix the (major, minor) if it has changed
-		if err := sys.MakeBlockFile(directCSIDevice, existingDrive.Status.MajorNumber, existingDrive.Status.MinorNumber); err != nil {
+		if err := sys.MakeBlockFile(mountSource, existingDrive.Status.MajorNumber, existingDrive.Status.MinorNumber); err != nil {
 			return err
 		}
-
 		mountTarget := filepath.Join(sys.MountRoot, existingDrive.Status.FilesystemUUID)
-		// Check if the drive is mounted
-		isMounted := false
-		for _, mount := range d.mounts {
-			if mount.MountSource == directCSIDevice {
-				isMounted = true
-				break
+
+		major, minor, err := sys.GetMajorMinor(mountSource)
+		if err != nil {
+			return err
+		}
+
+		mounted := false
+		if mounts, found := d.mounts[fmt.Sprintf("%v:%v", major, minor)]; found {
```

Comment:
hide the string construction behind a struct method

```
d.getMountInfo(major, minor int) []sys.MountInfo
```

Reply:
Done